src/VecSim/spaces/IP/IP_SSE4_SQ8.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+ */
+#include "VecSim/spaces/space_includes.h"
+#include <iostream>
+#include <string.h>
+
+static inline void InnerProductStep(const float *&pVect1, const uint8_t *&pVect2, __m128 &sum_prod,
+                                    const __m128 &min_val_vec, const __m128 &delta_vec) {
+    // Load 4 float elements from pVect1
+    __m128 v1 = _mm_loadu_ps(pVect1);
+    pVect1 += 4;
+
+    // Load 4 uint8 elements from pVect2, convert to int32, then to float
+    __m128i v2_i = _mm_cvtepu8_epi32(_mm_castps_si128(_mm_load_ss((float *)pVect2)));
```

Comment:
Have you considered using _mm_cvtsi32_si128? 
OPT1: Something like _mm_cvtepu8_epi32(_mm_cvtsi32_si128(*pVect2))?
Even though I'm not sure that it'll be faster then _mm_load_ss it might be worth to check.

I also see there is _mm_cvtpu8_ps (4Xu8 -> 4Xf32) which takes _m64. 
So how about:
v2_f = _mm_cvtpu8_ps(_mm_movepi64_pi64(mm_cvtsi32_si128(*pVect2)))